src/main/java/com/monitorjbl/plugins/UserUtils.java

```diff
@@ -19,44 +23,44 @@ public UserUtils(UserService userService) {
     this.userService = userService;
   }
 
-  public List<User> dereferenceUsers(Iterable<String> users) {
-    List<User> list = newArrayList();
-    for(String u : users) {
-      Optional<User> user = getUserByName(u);
-      if(user.isPresent()) {
-        list.add(user.get());
-      }
+  public List<User> dereferenceUsers(Collection<String> usernames) {
+    Set<String> distinctNames = new HashSet<>(usernames);
+    if (distinctNames.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    Set<ApplicationUser> usersByName = userService.getUsersByName(distinctNames);
+    if (usersByName.isEmpty()) {
+      return Collections.emptyList();
     }
-    return list;
-  }
 
-  public Optional<User> getUserByName(String username) {
-    ApplicationUser user = userService.getUserByName(username);
-    return user == null ? Optional.empty() : Optional.of(new User(user.getName(), user.getDisplayName()));
+    return usersByName.stream()
+        .map(user -> new User(user.getName(), user.getDisplayName()))
+        .collect(toList());
   }
 
   public String getUserDisplayNameByName(String username) {
-	    ApplicationUser user = userService.getUserByName(username);  
-	    return user.getDisplayName() == null ? username : user.getDisplayName();
+    return ofNullable(userService.getUserByName(username))
+        .map(ApplicationUser::getDisplayName)
+        .orElse(username);
   }
   
   public ApplicationUser getApplicationUserByName(String username) {
     return userService.getUserByName(username);
   }
 
-  public List<String> dereferenceGroups(List<String> groups) {
-    List<String> users = newArrayList();
-    for(String group : groups) {
-      List<ApplicationUser> results = newArrayList(userService.findUsersByGroup(group, new PageRequestImpl(0, RESULTS_PER_REQUEST)).getValues());
-      for(int i = 1; results.size() > 0; i++) {
-        users.addAll(results.stream()
-            .map(ApplicationUser::getSlug)
-            .collect(Collectors.toList()));
-        results = newArrayList(userService.findUsersByGroup(group, new PageRequestImpl(i * RESULTS_PER_REQUEST, RESULTS_PER_REQUEST)).getValues());
```

Comment:
So the code was requesting 25 users at a time. If 24 were returned (meaning there wasn't a 25th, and indicating there are no more users), because `results.size()` would still be `>0`, you'd do an extra page load trying to go "beyond" the end. When _that_ returned empty, you'd stop.